### ISSUE TITLE
Add type "HTTP" for resource aws_appsync_datasource

### ISSUE THREAD

**Issue by vitali-ausianik** [terraform-providers/terraform-provider-aws#5674]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

For now this provider supports only four types of AppSync resolvers - Lambda, DynamoDB, ElasticSearch, None.
But AWS supports one more - HTTP.
Proof: https://docs.aws.amazon.com/cli/latest/reference/appsync/create-data-source.html

### New or Affected Resource(s)

* aws_appsync_datasource

### Potential Terraform Configuration

```hcl
resource "aws_appsync_datasource" "datasource" {
  api_id = "someApiId"
  name = "someName"
  type = "HTTP"
  http_config {
    endpoint = "https://api.domain.com"
  }
}
```

**Comment by bflad** [comment]
Support for HTTP type and configuration has been merged and will be released with version 1.41.0 of the AWS provider, likely later today. 👍

**Comment by bflad** [comment]
This has been released in [version 1.41.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1410-october-18-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
